parsl/curvezmq.py

```diff
@@ -144,7 +144,7 @@ def _start_auth_thread(self) -> ThreadAuthenticator:
         auth_thread.start()
         # Only allow certs that are in the cert dir
         assert self.cert_dir  # For mypy
-        auth_thread.configure_curve(domain="*", location=self.cert_dir)
+        auth_thread.configure_curve(domain="*", location=str(self.cert_dir))
```

Comment:
what's this for?

Reply:
I was getting an error from mypysince we allow cert_dir to be Pathlike, but configure_curve wants a string. I could make this a separate PR if needed.

Reply:
ok. that definitely should be separate from this PR - @rjmello wrote that bit and might have some thoughts.

Are you using a different version of mypy compared to parsl CI?

Reply:
The `location` argument accepts a `str` or `PathLike` object (see [pyzmq code](https://github.com/zeromq/pyzmq/blob/01bd01c77277f16f714807a3ae4769f5b726710a/zmq/auth/base.py#L134)). This sounds like a mypyism.

Reply:
This is not true for all versions of pyzmq. I am running Ubuntu 24.04 with Python 3.12.6, with a system installed pyzmq (python3-zmq for Ubuntu) 24.0.1 In this version the `configure_curve` call maps to [here](https://github.com/zeromq/pyzmq/blob/6ff49f843f1f4932422aa68f4cc286fbd7e9e203/zmq/auth/thread.py#L215) where the location argument accepts only a `str`

I will remove this change from this PR, but I think this will need to be looked at more deeply.

Reply:
That's a fair point since we officially support [pyzmq>=17.1.2](https://github.com/Parsl/parsl/blob/7241f29a293c23453ef1880691f90a2bda4c58c3/requirements.txt#L1), which is older than your version and accepts the [same type](https://github.com/zeromq/pyzmq/blob/027c800ddf63135a6d706841edb3c2b44ac43a37/zmq/auth/thread.py#L177). I'm personally fine with this change and maybe a brief comment.